ddev/ddev
# fix: Don't kill supervisord when processes exit, for #5906, fixes #5358

## The Issue

- #5358
- #5906
- #1142

## How This PR Solves The Issue

The logic added in #1142 was done before healthchecks were added to Docker containers.

The change proposed in #5906 makes the logic even more complex.

Alternatively, we can remove `kill_supervisor.py` completely.

## Manual Testing Instructions

Use this in `.ddev/config.yaml`:

```
web_extra_daemons:
  - name: baddaemon
    command: somemissingcommand
    directory: /var/www/html
  - name: gooddaemon
    command: 'while true; do date; sleep 60; done'
    directory: /var/www/html
```

```
ddev logs # see correct behavior for baddaemon
WARN exited: baddaemon (exit status 127; not expected)
```

```
ddev exec killall php-fpm
ddev logs # see it restarting again
INFO success: php-fpm entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
```

## Automated Testing Overview



## Related Issue Link(s)

## Release/Deployment Notes
